packages/@webex/plugin-cc/test/unit/spec/services/core/WebSocket/WebSocketManager.ts

```diff
@@ -0,0 +1,177 @@
+      return 'blob:http://localhost:3000/12345';
+    };
+
+    webSocketManager = new WebSocketManager({ webex: mockWebex });
+
+    setTimeout(() => {
+      MockWebSocket.inst.onopen();
+      MockWebSocket.inst.onmessage({ data: JSON.stringify({ type: "Welcome" }) });
+      webSocketManager.close(false);
+    }, 1);
+
+    console.log = jest.fn();
+    console.error = jest.fn();
+  });
+
+  it('should initialize WebSocketManager', () => {
+    expect(webSocketManager).toBeDefined();
+  });
+
+  it('should register and connect to WebSocket', async () => {
+    const subscribeResponse = {
+      body: {
+        webSocketUrl: 'wss://fake-url',
+      },
+    };
+
+    (mockWebex.request as jest.Mock).mockResolvedValueOnce(subscribeResponse);
+
+    const subscribeRequest: SubscribeRequest = {
+      routingId: 'test-routing-id',
+    };
+
+    await webSocketManager.initWebSocket({ body: subscribeRequest });
+
+    expect(mockWebex.request).toHaveBeenCalledWith({
+      service: WCC_API_GATEWAY,
+      resource: SUBSCRIBE_API,
+      method: 'POST',
+      body: subscribeRequest,
+    });
+  });
+
+  it('should close WebSocket connection', async () => {
+    const subscribeResponse = {
+      body: {
+        webSocketUrl: 'wss://fake-url',
+      },
+    };
+
+    (mockWebex.request as jest.Mock).mockResolvedValueOnce(subscribeResponse);
+
+    const subscribeRequest: SubscribeRequest = {
+      routingId: 'test-routing-id',
+    };
+
+    await webSocketManager.initWebSocket({ body: subscribeRequest });
+
+    webSocketManager.close(true, 'Test reason');
+
+    expect(MockWebSocket.inst.close).toHaveBeenCalled();
+    expect(mockWorker.postMessage).toHaveBeenCalledWith({ type: 'terminate' });
+  });
+
+  it('should handle WebSocket keepalive messages', async () => {
+    const subscribeResponse = {
+      body: {
+        webSocketUrl: 'wss://fake-url',
+      },
+    };
+
+    (mockWebex.request as jest.Mock).mockResolvedValueOnce(subscribeResponse);
+
+    const subscribeRequest: SubscribeRequest = {
+      routingId: 'test-routing-id',
+    };
+
+    await webSocketManager.initWebSocket({ body: subscribeRequest });
+
+    setTimeout(() => {
+      MockWebSocket.inst.onopen();
+      MockWebSocket.inst.onmessage({ data: JSON.stringify({ type: 'keepalive' }) });
+      mockWorker.postMessage({
+        data: {
+          type: 'keepalive'
+        }
+      });
+    }, 1);
+
+    expect(MockWebSocket.inst.send).toHaveBeenCalledWith(JSON.stringify({ keepalive: 'true' }));
+  });
+
+  it('should handle WebSocket close due to network issue', async () => {
+    const subscribeResponse = {
+      body: {
+        webSocketUrl: 'wss://fake-url',
+      },
+    };
+
+    (mockWebex.request as jest.Mock).mockResolvedValueOnce(subscribeResponse);
+
+    const subscribeRequest: SubscribeRequest = {
+      routingId: 'test-routing-id',
+    };
+
+    await webSocketManager.initWebSocket({ body: subscribeRequest });
+
+    setTimeout(() => {
```

Comment:
We need to set this timeout so that we cna call the `onopen` callback and the code flows as per usual.